pkg/cloud/services/eks/cluster.go

```diff
@@ -550,6 +573,29 @@ func (s *Service) describeEKSCluster(eksClusterName string) (*eks.Cluster, error
 	return out.Cluster, nil
 }
 
+func (s *Service) updateEncryptionConfig(updatedEncryptionConfigs []*eks.EncryptionConfig) error {
+	input := &eks.AssociateEncryptionConfigInput{
+		ClusterName:      aws.String(s.scope.KubernetesClusterName()),
+		EncryptionConfig: updatedEncryptionConfigs,
+	}
+	if err := wait.WaitForWithRetryable(wait.NewBackoff(), func() (bool, error) {
+		if _, err := s.EKSClient.AssociateEncryptionConfig(input); err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				return false, aerr
+			}
+			return false, err
+		}
+
+		conditions.MarkTrue(s.scope.ControlPlane, eks.UpdateTypeAssociateEncryptionConfig)
```

Comment:
I like including a condition. But this will require some additional code to ensure we mark false as well. When we call `AssociateEncryptionConfig` what status will the EKS cluster change to?

I'm guessing it migt change to: `ClusterStatusUpdating = "UPDATING"`

As an example of how we use existing conditions [EKSControlPlaneUpdatingCondition](https://github.com/kubernetes-sigs/cluster-api-provider-aws/search?q=EKSControlPlaneUpdatingCondition) is good to look at.

Reply:
I am not sure what sttaus will EKSCluster change to, in what case we should mark false?

Reply:
do we want to add something like [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/e55d3ed297b9608aa229cf96caa658e64ef6373d/pkg/cloud/services/eks/cluster.go)

Reply:
I'm thinking that we may need to change the condition to `EKSControlPlaneUpdatingCondition` as well to indicate that an update is in progress.....assuming that the `AssociateEncryptionConfig` call results in the status of the cluster updating.

Reply:
I'll have a quick look at the status now....

Reply:
The status of the cluster does change to to "UPDATING":
![image](https://user-images.githubusercontent.com/198425/122750261-30bbd980-d286-11eb-96cb-bcbdd338170e.png)